### ISSUE TITLE
Easier introspection of jsonb fields

### ISSUE THREAD

**Issue by onomated** [jrochkind/attr_json#46]
I'm looking to dynamically generate graphql type definitions based on fields in my model classes.  I can instrospect ActiveRecord attributes via:

```ruby
def Foo < ActiveRecord::Base
  include AttrJson::Record
end

# columns
Foo.columns
# associations
Foo.reflect_on_all_associations
```

I can access `Foo.attr_json_registry` but currently have no easy way to deduce json attribute names.  
Currently hacking around to get the attributes, something around the lines of:

```ruby
      def attr_json_fields(model)
        return unless model < AttrJson::Record

        registry = model.attr_json_registry
        attrs = model.instance_methods.select { |m| m.to_s.ends_with? '=' }
        attrs.map! { |attr| attr.to_s[0...-1] }
        attrs.select { |attr| registry.has_attribute? attr }
      end
```

Any objections to a `fields` method in the `AttrJson::AttributeDefinition::Registry` class?

**Comment by jrochkind** [comment]
Hmm, you just want to get the list of all attr_json registered fields?

Yes, you should be able to do that off the registry!  If you can't right now, it's an oversight. PR very welcome! I'm going to need this myself soon too probably.

**Comment by jrochkind** [comment]
Not totally sure if `fields` is the right name for the method, I'm not sure if we use the term "fields" anywhere else?  Maybe, uh.... `registered_attributes`?  Not sure.

**Comment by onomated** [comment]
Yup, `registered_attributes` sounds good/intuitive.  Yes its as simple as:

```ruby
def registered_attributes
  @name_to_definition.keys
end
```

I can work on a PR for that soon, once I get some deliverables completed.

**Comment by jrochkind** [comment]
I need this soon, so will go ahead and implement myself today!
